packages/zend-loader/library/Zend/Loader/ClassMapAutoloader.php

```diff
@@ -237,7 +237,7 @@ public static function concatPharParts($part)
      * @param  array $parts 
      * @return void
      */
-    public static function resolvePharParentPath($value, $key, &$parts)
+    public static function resolvePharParentPath($value, $key, $parts)
```

Comment:
I am not sure here, since the method does not return anything, but modifies $parts inside - I believe the reference must stay here. Does PHP 8 complain here?

Reply:
PHP8 does complain that $parts is value when the function is called from `realPharPath`. I assumed PHP8 had changed something fundamentally about pass-by-ref. Turns out that it's only `array_walk` that's changed and it now automatically passes the $userdata param by value meaning that this can be circumvented by changing array_walk to a foreach loop.